### ISSUE TITLE
"runCommands" command doesn't work with "ctrl+k" keyboard shortcut

### ISSUE THREAD

**Issue by mklement0** [microsoft/vscode#179724]
Type: <b>Bug</b>

Trying to define a "ctrl+k" keyboard shortcut that uses the new "runCommands" command to run _multiple_ commands is ineffective.

Using a _different character_ - e.g. "ctrl+1" or adding "shift" ("ctrl+shift+k") makes the problem go away.

The following two definitions _should_ be equivalent, but only the single-command  definition works (sends a `"hi from ..."` string to the terminal):

*  Functioning single-command definition:

```json
  {
    "key": "ctrl+k",
    "command": "workbench.action.terminal.sendSequence",
    "args": {
      "text": "hi from single command"
    }
  }
```

*  Equivalent, but broken multi-command definition:

```json
  {
    "key": "ctrl+k",
    "command": "runCommands",
    "args": {
      "commands": [
        {
          "command": "workbench.action.terminal.sendSequence",
          "args": {
            "text": "hi from runCommands"
          }
        }
      ]
    }
  }
```

Observed on macOS 13 and Windows 11 (haven't tried Linux)

The symptom on macOS is that nothing happens, and on Windows it _prints_ `^K` to the terminal instead of sending the string.

**Note**: If you switch away from the terminal tab to, say, `OUTPUT`, the problem goes away (that is, the string is sent while the terminal is _hidden_).

Output from the troubleshooting tool when the bug _does_  manifest (terminal visible); note the `Keyboard event cannot be dispatched` lines:

```none
2023-04-11 17:40:53.652 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-11 17:40:53.652 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-04-11 17:40:53.652 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 17, key: Control
2023-04-11 17:40:53.653 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 5 ('Ctrl')
2023-04-11 17:40:53.653 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2023-04-11 17:40:53.809 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-11 17:40:53.810 [info] [KeybindingService]: | Resolving ctrl+[KeyK]
2023-04-11 17:40:53.810 [info] [KeybindingService]: \ From 2 keybinding entries, matched runCommands, when: no when condition, source: user.
2023-04-11 17:40:53.971 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-11 17:40:53.972 [info] [KeybindingService]: | Resolving ctrl+[KeyK]
2023-04-11 17:40:53.972 [info] [KeybindingService]: \ From 2 keybinding entries, matched runCommands, when: no when condition, source: user.
2023-04-11 17:40:54.132 [info] [KeybindingService]: / Soft dispatching keyboard event
2023-04-11 17:40:54.132 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2023-04-11 17:40:54.133 [info] [KeybindingService]: + Ignoring single modifier ctrl due to it being pressed together with other keys.
```

VS Code version: Code 1.77.2 (Universal) (e344f1f539a80912a0e9357cec841f36ce97a4e2, 2023-04-06T10:12:31.126Z)
OS version: Darwin arm64 22.3.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|7, 6, 6|
|Memory (System)|32.00GB (0.15GB free)|
|Process Argv|/Users/mklement/Desktop/pg/pg.ps1 --crash-reporter-id 2090eeb5-14c6-4ea3-bec5-b0fb25907629|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by Tyriar** [comment]
ctrl+k is a special keybinding due to all the chords on it. You should be able to workaround this by disabling allowChords https://code.visualstudio.com/docs/terminal/advanced#_chords

**Comment by mklement0** [comment]
`ctrl+k` is _not_ special on macOS -  there it is `cmd-k` that starts _chords_.

* In fact, If you use `cmd-k` instead of `ctrl-k` on macOS, the `runCommands` definition _does_ work.

* Conversely, what you're suggesting to make `ctrl-k` work - adding `"terminal.integrated.allowChords": true` to `settings.json` - does NOT work, neither on macOS nor on Windows.

As stated, the definition _does_ work with a single command, and malfunctions _only_ with `runCommands` _while in the terminal_ (even when limiting the scope to the terminal with `"when": "terminalFocus"`, which is advisable on Windows, so as not to preempt _chords_ in an editor tab, for instance).

In short:

* There is clearly a bug.

* The suggested workaround isn't one.

Please reopen this issue.

**Comment by Tyriar** [comment]
Oh I missed the mac part.

This is happening because `runCommands` is not in the default [commands to skip shell](https://code.visualstudio.com/docs/terminal/advanced#_keybinding-and-the-shell). I'll reopen to add that.

In the meantime you can workaround it via `"terminal.integrated.commandsToSkipShell": ["runCommands"]`

**Comment by mklement0** [comment]
Thank you, @Tyriar. I appreciate the (temporary) reopening and the fix.
